hooks/useBalances.ts

```diff
@@ -1,27 +1,66 @@
-import { useAccount, useBalance, useNetwork } from 'wagmi';
+import { useQueryClient } from '@tanstack/react-query';
+import { useEffect } from 'react';
+import { erc20Abi, formatUnits } from 'viem';
+import {
+  useAccount,
+  useBalance,
+  useBlockNumber,
+  useReadContracts,
+} from 'wagmi';
 import { wethAddrs } from '../utils/contracts';
 
 const useBalances = () => {
-  const { address } = useAccount();
-  const { chain } = useNetwork();
+  const { address, chain } = useAccount();
+  const contractAddress = wethAddrs?.[chain?.name.toLowerCase() || 'homestead'];
+  const queryClient = useQueryClient();
 
-  const contractAddress = wethAddrs?.[chain?.network || 'homestead'];
+  const { data: blockNumber } = useBlockNumber({ watch: true });
 
-  const getEthBalance = useBalance({
+  const { data: ethBalanceData, queryKey: ethQueryKey } = useBalance({
     address,
-    enabled: contractAddress?.length !== 0,
-    watch: true,
+    query: {
+      enabled: !!address && !!contractAddress,
+    },
   });
 
-  const getWethBalance = useBalance({
-    address,
-    enabled: contractAddress?.length !== 0,
-    watch: true,
-    token: contractAddress,
+  const { data: wethBalanceData, queryKey: wethQueryKey } = useReadContracts({
+    contracts: address
+      ? [
+          {
+            address: contractAddress || '',
+            abi: erc20Abi,
+            functionName: 'balanceOf',
+            args: [address],
+          },
+          {
+            address: contractAddress || '',
+            abi: erc20Abi,
+            functionName: 'decimals',
+          },
+        ]
+      : [],
+    query: {
+      enabled: !!address && !!contractAddress,
+    },
   });
 
-  const ethBalance = getEthBalance.data?.formatted || '0';
-  const wethBalance = getWethBalance.data?.formatted || '0';
+  useEffect(() => {
+    if (blockNumber) {
+      queryClient.invalidateQueries({ queryKey: ethQueryKey });
```

Comment:
smart but maybe overkill. the general query key (for useReadContracts) is `readContracts` which will likely clear all your current queries.

Reply:
I'd like to think so too but I'm just following the [docs](https://wagmi.sh/react/guides/migrate-from-v1-to-v2#removed-watch-property)